src/SprykerEco/Zed/CrefoPay/Business/Reader/CrefoPayReader.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\CrefoPay\Business\Reader;
+
+use Generated\Shared\Transfer\CrefoPayToSalesOrderItemsCollectionTransfer;
+use Generated\Shared\Transfer\CrefoPayToSalesOrderItemTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayOrderItemCollectionTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayOrderItemToCrefoPayApiLogTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayOrderItemToCrefoPayNotificationTransfer;
+use Generated\Shared\Transfer\PaymentCrefoPayTransfer;
+use SprykerEco\Zed\CrefoPay\Persistence\CrefoPayRepositoryInterface;
+
+class CrefoPayReader implements CrefoPayReaderInterface
+{
+    protected const SUCCESS_RESULT_CODES = [0, 1];
+
+    /**
+     * @var \SprykerEco\Zed\CrefoPay\Persistence\CrefoPayRepositoryInterface
+     */
+    protected $repository;
+
+    /**
+     * @param \SprykerEco\Zed\CrefoPay\Persistence\CrefoPayRepositoryInterface $repository
+     */
+    public function __construct(CrefoPayRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @param int $fkSalesOrder
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayTransfer
+     */
+    public function findPaymentCrefoPayByIdSalesOrder(int $fkSalesOrder): PaymentCrefoPayTransfer
```

Comment:
Could you rename variable and make it w/o fk prefix?

Reply:
Fixed.